hyperjumptech/monika
# added: self update

# Monika Pull Request (PR)  
This PR resolves #558 

## What feature/issue does this PR add  
1. Add flag `--auto-update major|minor|patch` for automatic update Monika
2. Add documentation for `--auto-update` flag

## How did you implement / how did you fix it  
1. Create new plugin `src/plugins/updater/index.ts`
2. Support update for various Monika installations, i.e. npm, tarball generated by `oclif pack` command, and binary packaged by `vercel/pkg`
4. Checksum for downloaded tarballs.

## How to test  
1. Testing for npm installation
    1. Edit version in `package.json` file, e.g. `1.6.12`
    2. `npm pack`
    3. `npm install -g hyperjumptech-monika-*.tgz`
    5. `monika --auto-update major -c monika.example.yml`
    6. Should see log beginning with `Updater`
    7. After terminated itself, verify Monika new version by running `monika -v`
2. Testing for binary installation
    1. Edit version in `package.json` file, e.g. `1.6.12`
    2. `npm run pkg`
    3. `./dist/monika-linux -c monika.example.yml --auto-update major`. Use the binary depending on your platform.
    4. Should see log beginning with `Updater`
    5. After terminated itself, verify Monika new version by running `monika -v`

## Screenshot
![image](https://user-images.githubusercontent.com/9563873/157190633-be91c7df-5a88-4e71-8970-b7af5b9e534d.png)
